modules/migrations/github.go

```diff
@@ -116,8 +132,18 @@ func (g *GithubDownloaderV3) SetContext(ctx context.Context) {
 }
 
 func (g *GithubDownloaderV3) sleep() {
-	for g.rate != nil && g.rate.Remaining <= GithubLimitRateRemaining {
-		timer := time.NewTimer(time.Until(g.rate.Reset.Time))
+	var recentIdx int
+	var maxRemaining int
+	for i := 0; i < len(g.clients); i++ {
+		if g.rates[i] != nil && g.rates[i].Remaining > maxRemaining {
+			maxRemaining = g.rates[i].Remaining
+			recentIdx = i
+		}
+	}
```

Comment:
Is there an edging case? eg: if all client's remaining becomes 0, then this function seems to only refresh the first client's rate. Do other clients still have a chance to get a rate refresh?